prov/efa/src/efa_hmem.c

```diff
@@ -181,6 +188,79 @@ static inline void efa_hmem_info_check_p2p_support_cuda(struct efa_hmem_info *in
 	return;
 }
 
+static inline void efa_hmem_info_check_p2p_support_rocr(struct efa_hmem_info *info) {
+#if HAVE_ROCR
+	void *ptr = NULL;
+	struct ibv_mr *ibv_mr;
+	struct ibv_pd *ibv_pd;
+	int ibv_access = IBV_ACCESS_LOCAL_WRITE | IBV_ACCESS_REMOTE_READ;
+	size_t len = ofi_get_page_size() * 2;
+	int ret;
+	int dmabuf_fd;
+	uint64_t dmabuf_offset;
+
+	ptr = rocr_alloc(len);
+	if (!ptr) {
+		info->initialized = false;
+		EFA_WARN(FI_LOG_CORE, "Failed to allocate ROCr buffer\n");
+		return;
+	}
+
+	ibv_pd = ibv_alloc_pd(g_efa_selected_device_list[0].ibv_ctx);
+	if (!ibv_pd) {
+		EFA_WARN(FI_LOG_CORE, "Failed to allocate ibv_pd: %d\n", errno);
+		rocr_free(ptr);
+		return;
+	}
+
+#if HAVE_EFA_DMABUF_MR
+	ret = rocr_hmem_get_dmabuf_fd(ptr, len, &dmabuf_fd, &dmabuf_offset);
+	if (ret == FI_SUCCESS) {
+		ibv_mr = ibv_reg_dmabuf_mr(ibv_pd, dmabuf_offset,
+					   len, (uint64_t) ptr, dmabuf_fd, ibv_access);
+		(void) rocr_hmem_put_dmabuf_fd(dmabuf_fd);
```

Comment:
I didn't see dmabuf ops in rocr_hmem.c. Rocr doesn't support it? That means you will always try the side channel, and that likely won't work unless the efa driver implemented it for rocr

Reply:
Sorry for the confusion. Now I can see rocr_hmem_{get,put}_dmabuf_fd() is implemented in src/hmem_rocr.c, and not in fabtests code yet

However, this PR doesn't seem to go this code path for the fi_mr_regattr via a VA (no FI_MR_DMABUF) as it finally calls ibv_reg_mr

The expected workflow would be 

1. efa_mr_reg_ibv_mr should retrieve the dmabuf fd internally and reg via ibv_reg_dmabuf_mr, when dmabuf is available
2. fabtests implement that dmabuf code as well and it will call fi_mr_regattr with FI_MR_DMABUF flags and fd directly